whd/dispatcher.py

```diff
@@ -211,6 +213,106 @@ def _set_pr_labels(self, pr_event, resources):
                     f" in '{repository_path}' found. Ignoring, since they are not an org member'."
                 )
 
+    def _start_not_triggered_jobs(self, concourse_api, resources):
+        '''
+        check all PR resource versions for which no build exists. If a resource version has
+        no build, pin this resource version, trigger the corresponding job and unpin the resource
+        '''
+        for resource in resources:
+            if resource.type != "pull-request":
+                continue
+
+            resource_versions = concourse_api.resource_versions(
+                pipeline_name=resource.pipeline.name,
+                resource_name=resource.name
```

Comment:
as noted above: we know the specific PR for which we want to do this. so there is no need to process _all_ resources, is there?